Sources/MongoSwift/MongoClient.swift

```diff
@@ -187,9 +187,14 @@ public class MongoClient {
             throw UserError.invalidArgumentError(message: "libmongoc not built with TLS support")
         }
 
+        self.encoder = BSONEncoder(options: options)
+        self.decoder = BSONDecoder(options: options)
+
         // if a readConcern is provided, set it on the client
         if let rc = options?.readConcern {
```

Comment:
now that we don't need to access `rc._readConcern` I think we could rewrite this using optional chaining as

`try options?.readConcern?.withMongocReadConcern { ....`

same applies for where you call the equivalent methods in `MongoDatabase` and `MongoCollection` initializers